src/gazebo_mavlink_interface.cpp

```diff
@@ -609,6 +609,29 @@ void GazeboMavlinkInterface::send_mavlink_message(const mavlink_message_t *messa
   }
 }
 
+template <class T>
+void GazeboMavlinkInterface::setSensorOrientation(ignition::math::Vector3d u_Xs, T sensor_msg) {
```

Comment:
sensor_msg.orientation needs to be passes as a reference otherwise it's never set

Reply:
Thanks @mrivi! That solves the problem!